src/dune_rules/compilation_context.ml

```diff
@@ -175,6 +178,16 @@ let create ~super_context ~scope ~expander ~obj_dir ~modules ~flags
     ; stdlib
     }
   in
+  let melange_js_includes =
+    let open Resolve.Memo.O in
+    Command.Args.memo
+      (Resolve.Memo.args
+         (let+ libs = requires_compile in
+          Command.Args.S
+            [ Lib_flags.L.include_flags ~project libs Melange
+            ; Hidden_deps (Lib_file_deps.deps libs ~groups:[ Melange Js ])
```

Comment:
I did not think adding `Cmi` and `Cmj` here is necessary, as the `Js` files in libs depend on their own `Cm*`.